UnityProject/Assets/Scripts/Items/Science/VacpackGun.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections;
+using System.Collections.Generic;
+using HealthV2;
+using Tilemaps.Behaviours.Layers;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+public class VacpackGun : MonoBehaviour, 	ICheckedInteractable<PositionalHandApply>
+{
+
+	[SerializeField] private ItemStorage storage;
+	public VacpackBackpack VacpackBackpack;
+
+	private void Start()
+	{
+		if (storage == null) storage = gameObject.PickupableOrNull().ItemSlot.ItemStorage;
+		VacpackBackpack = storage.GetComponent<VacpackBackpack>();
+
+	}
+
+/*
+	(
+		PlayerScript playerScript,
+		GameObject target,
+		NetworkSide side,
+		bool allowSoftCrit = false,
+		ReachRange reachRange = ReachRange.Standard,
+		Vector2? targetPosition = null,
+		Vector2? targetVector = null,
+		RegisterTile targetRegisterTile = null,
+		PlayerTypes apt = PlayerTypes.Normal
+	)
+	*/
+	public bool WillInteract(PositionalHandApply interaction, NetworkSide side)
+	{
+		if (Validations.CanInteract(interaction.PerformerPlayerScript,side, false) == false) return false;
+
+		var Distance = (interaction.Performer.AssumedWorldPosServer() - interaction.WorldPositionTarget.To3()).magnitude;
+		if (Distance > 2.5f) return false;
```

Comment:
why is this hardcoded? Wouldn't this be a problem for future buffs to interaction distances?

Reply:
I think it's out of scope for now, but definitely needs to be worked on at some point